android/src/main/java/com/ekreutz/barcodescanner/ui/BarcodeScannerView.java

```diff
@@ -223,6 +233,19 @@ public boolean setFocusMode(int focusMode) {
         return mCameraSource != null && mCameraSource.setFocusMode(PREFERRED_FOCUS_MODES[focusMode]);
     }
 
+    /**
+     * Sets torch mode.
+     * Possible values: 0 = off, 1 = torch (always on)
+     * @param torchMode
+     */
+    public boolean setTorchMode(int torchMode) {
+        if (torchMode < 0 || torchMode > 1) {
+            torchMode = 0;
+        }
+
+        return mCameraSource != null && mCameraSource.setFlashMode(RELEVANT_FLASH_MODES[torchMode]);
```

Comment:
On some devices FLASH_MODE_TORCH is not supported and the best way is to specify FLASH_MODE_ON.
if (mCameraSource != null) {
            if (torchMode == 0) {
                return mCameraSource.setFlashMode(Camera.Parameters.FLASH_MODE_OFF);
            }
            return mCameraSource.setFlashMode(Camera.Parameters.FLASH_MODE_TORCH)
                    || mCameraSource.setFlashMode(Camera.Parameters.FLASH_MODE_ON);
        }

Reply:
Ok, I'll update my pull request